### PR TITLE
[UI] EVEREST-1681 Add edit PITR modal to overview cards / disable edit buttons for deleting/restoring status

### PR DESCRIPTION
**Pull request by dianabirs**
[EVEREST-1681](https://perconadev.atlassian.net/browse/EVEREST-1681)

image
image

[EVEREST-1681]: https://perconadev.atlassian.net/browse/EVEREST-1681?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: ui/apps/everest/src/pages/db-cluster-details/cluster-overview/cards/backups-details.tsx [hunk 1]
```diff
@@ -172,6 +195,12 @@ export const BackupsDetails = ({
           dataTestId="pitr"
           title={Messages.titles.pitr}
           loading={loading}
+          actionButtonProps={{
+            onClick: () => {
+              setOpenEditModal(true);
+            },
+          }}
+          editable={canUpdateDb && !restoring && !deleting}
```
Comment (fabio-silva) [inline ui/apps/everest/src/pages/db-cluster-details/cluster-overview/cards/backups-details.tsx:203]: if the number of statuses to check gets bigger, we'll end up with lots of different booleans.
Something like:
```
//...
const editable = ![DbClusterStatus.restoring, DbClusterStatus.deleting].includes(dbCluster?.status?.status);
//...
editable={editable}
```
Might be more readable.

Comment (fabio-silva) [reply ui/apps/everest/src/pages/db-cluster-details/cluster-overview/cards/backups-details.tsx:203]: Same for the others similar below

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: ui/apps/everest/src/pages/db-cluster-details/cluster-overview/cards/backups-details.tsx (hunk 1 above)
Comment (dianabirs) [reply ui/apps/everest/src/pages/db-cluster-details/cluster-overview/cards/backups-details.tsx:203]: Agreed, but I would keep restoring and deleting as separate consts for better visibility, what do you think?

Comment (dianabirs) [reply ui/apps/everest/src/pages/db-cluster-details/cluster-overview/cards/backups-details.tsx:203]:
```
  const restoringOrDeleting = ![
    DbClusterStatus.restoring,
    DbClusterStatus.deleting,
  ].includes(dbCluster?.status?.status!);

  const editable = canUpdateDb && !restoringOrDeleting && canUpdateMonitoring;
```

Comment (fabio-silva) [reply ui/apps/everest/src/pages/db-cluster-details/cluster-overview/cards/backups-details.tsx:203]: sure, let's go with that

**Review by fabio-silva — APPROVED** [review]

**Pull request merged**